### PR TITLE
add complex support for  optest

### PR DESCRIPTION
**Pull request by GGBond8488**
### PR types
Others

### PR changes
Others 

### Description
Pcard-70652
add complex support for  optest

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/eager_op_test.py [hunk 1]
```diff
@@ -242,15 +250,55 @@ def __set_elem__(tensor, i, e):
...
+            # no input for dl/dy, though it should do. so there di will be zero
+            f_ajoint = np.ones_like(inputs).astype(tensor_to_check_dtype)
+            df_over_dr = (y_pos - y_neg) / delta / 2
+            df_over_di = (y_pos_j - y_neg_j) / delta / 2
+
+            dl_over_du, dl_over_dv = f_ajoint.real, f_ajoint.imag
+
+            du_over_dx, dv_over_dx = df_over_dr.real, df_over_dr.imag
+
+            du_over_dy, dv_over_dy = df_over_di.real, df_over_di.imag
```
Comment (jeff41404) [inline python/paddle/fluid/tests/unittests/eager_op_test.py]: `x` and `y` represent inputs and outputs, respectively, it is better to use `du_over_dr`, `du_over_di`, `dv_over_dr`, `dv_over_di` ?

Comment (GGBond8488) [reply python/paddle/fluid/tests/unittests/eager_op_test.py:288]: Here x and y are x, y in l = u(x,y) + iv(x,y)

File: python/paddle/fluid/tests/unittests/eager_op_test.py [hunk 2]
```diff
@@ -242,15 +250,55 @@ def __set_elem__(tensor, i, e):
...
+
+            dl_over_dr = np.sum(
+                dl_over_du * du_over_dx + dl_over_dv * dv_over_dx
+            )
+            dl_over_di = np.sum(
+                dl_over_du * du_over_dy + dl_over_dv * dv_over_dy
+            )
```
Comment (jeff41404) [inline python/paddle/fluid/tests/unittests/eager_op_test.py]: same problem as L286/L288

File: python/paddle/fluid/tests/unittests/eager_op_test.py (hunk 1 above)
Comment (GGBond8488) [reply python/paddle/fluid/tests/unittests/eager_op_test.py]: changed to du_over_dr, du_over_di, dv_over_dr, dv_over_di

File: python/paddle/fluid/tests/unittests/eager_op_test.py [hunk 3]
```diff
@@ -242,15 +250,55 @@ def __set_elem__(tensor, i, e):
...
+            x_neg_j = origin - 1j * delta
+            __set_elem__(tensor_to_check, i, x_neg_j)
+            y_neg_j = get_output()
+
+        __set_elem__(tensor_to_check, i, origin)
+
+        if tensor_to_check_dtype in [np.complex64, np.complex128]:
+            # always assume real output, because this function has
+            # no input for dl/di, though it should do. so there di will be zero
+            f_ajoint = np.ones_like(inputs).astype(tensor_to_check_dtype)
```
Comment (jeff41404) [inline python/paddle/fluid/tests/unittests/eager_op_test.py]: why use shape of inputs instead of shape of outputs

**Review by chenwhql — APPROVED** [review]

**Review by jeff41404 — APPROVED** [review]
LGTM

**Pull request merged**